git_md_page/forms/git_plugins.py

```diff
@@ -1,7 +1,36 @@
 """Forms for git related plugins."""
+from shutil import rmtree
+from tempfile import mkdtemp
+
 from django import forms
+from django.utils.translation import gettext as _
+from git import CommandError, Repo
+
+from git_md_page.models import GitRepository, GitTextPluginModel
+
+
+class GitRepositoryForm(forms.ModelForm):
+    """Form for GitRepository model."""
+
+    class Meta:
+        """Meta class."""
+
+        fields = "__all__"
+        model = GitRepository
+
+    def clean(self):
+        """Check repository existence."""
+        data = super().clean()
+
+        temp_folder = mkdtemp()
+        try:
+            Repo.clone_from(data["URL"], temp_folder, env={"GIT_TERMINAL_PROMPT": "0"}, depth=1)
+        except CommandError:
+            self.add_error("URL", _("Repository could not be cloned."))
```

Comment:
Agree, but

```
Cmd('git') failed due to: exit code(128)
  cmdline: git clone -v --depth=1 https://github.com/CZ-NIC/djangocms-git-md-page2.git /tmp/tmpikhbuetd
  stderr: 'Cloning into '/tmp/tmpikhbuetd'...
fatal: could not read Username for 'https://github.com': terminal prompts disabled
```

is not really user friendly. The last line is continuation of the previous `stderr` output, it is not possible to receive that part from an attribute from the `CommandError`. The only thing I am able to get is the error code, `128`. I can add it into the error message, but it will not help to the user.

Reply:
I suggest displaying the whole error this way:
```python
except CommandError as error:
    self.add_error("URL", str(error))
```

Reply:
Fixed.